### PR TITLE
docs(menu,popover): add isLazy prop for menu and popover

### PR DESCRIPTION
**Pull request by Zaynex**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes
      /start features)

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Fixes: [ISSUE NUMBER]

## What is the new behavior?

- add isLazy prop for `Menu` component

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
related issue #2146

### CODE REVIEW

File: website/pages/docs/overlay/menu.mdx [hunk 1]
```diff
@@ -197,6 +197,31 @@ component.
 </Menu>
 ```
 
+### Lazily mounting MenuItem
+
+By default, the `Menu` component renders all `MenuItem` to the DOM, meaning that invisible menu items are still rendered but are hidden by styles.
```
Comment (ljosberinn) [inline website/pages/docs/overlay/menu.mdx:202]: I'd like to rephrase this to `renders all children of "MenuList"` as there are not only `MenuItem`s that could be within a `MenuList`. There are also: `MenuGroup`, `MenuDivider`, `MenuOptionGroup` and `MenuItemOption`.

Same goes for below.

Comment (Zaynex) [reply website/pages/docs/overlay/menu.mdx:202]: By default, the `Menu` component renders all children of "MenuList" to the DOM, meaning that invisible children of "MenuList" are still rendered but are hidden by styles.

If you want to defer rendering of each children of "MenuList" until that menu is open, you can use the `isLazy` prop. This is useful if your menus require heavy performance, or make network calls on mount that should only happen when the component is displayed.

I replace all menu item with  children of "MenuList" . Is that ok ?

Comment (ljosberinn) [reply website/pages/docs/overlay/menu.mdx:202]: Sounds good! Please don't forget the `Popover` part, just making sure it isn't overread :)

Comment (Zaynex) [reply website/pages/docs/overlay/menu.mdx:202]: OK, just waiting for a song time.

Comment (Zaynex) [reply website/pages/docs/overlay/menu.mdx]: Please check it agian.

**Comment by ljosberinn** [comment]
@allcontributors please add @Zaynex for docs
